public/app/features/transformers/editors/EnumMappingRow.tsx

```diff
@@ -0,0 +1,160 @@
+import { css } from '@emotion/css';
+import React, { FormEvent, useState, KeyboardEvent, useRef, useEffect } from 'react';
+import { Draggable } from 'react-beautiful-dnd';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, Input, IconButton, HorizontalGroup, FieldValidationMessage, useStyles2 } from '@grafana/ui';
+
+type EnumMappingRowProps = {
+  convertFieldTransformIndex: number;
+  value: string;
+  index: number;
+  mappedIndex: number;
+  onChangeEnumValue: (index: number, value: string) => void;
+  onRemoveEnumRow: (index: number) => void;
+  checkIsEnumUniqueValue: (value: string) => boolean;
```

Comment:
should this be a passed-in prop or can we just directly use the `checkIsEnumUniqueValue` from https://github.com/grafana/grafana/pull/76410/files#diff-0f946700206cc388640fbe8e8df18c407f2a947e1b8bd5a0acca1b51dc91477aR102-R104?

dunno if you had concrete plans for more than one pluggable strategy for uniquness testing?

Reply:
main reason for `checkIsEnumUniqueValue` being passed as a prop is wanting to avoid sending the entire `enumRows` object as a prop to every instance of the `EnumMappingRow` component